src/python/grpcio/commands.py

```diff
@@ -58,6 +63,117 @@ class CommandError(Exception):
   """Simple exception class for GRPC custom commands."""
 
 
+# TODO(atash): Remove this once PyPI has better Linux bdist support
+def _linux_target_egg_name(grpc_version):
+  """Get gRPC's custom made-up target names for Linux."""
+  assert sys.platform.startswith('linux')
+  version = '.'.join(map(str, sys.version_info[0:2]))
+  architecture = platform.machine()
```

Comment:
My suspicion is that platform.machine() won't work on 32bit.

On my machine, it returns "x86_64" and when I run it under docker image 32bit\debian:jessie it still returns "x86_64" (because that's my machine still).

Consider incorporating platform.architecture()[0]  which returns "32bit" and "64bit" correctly. Also, be aware that the doc says that this doesn't necessarily work on MacOS where you can have a fat binary. There you might use sys.maxsize (that won't work on Windows in turn because sys.maxsize is only 32bit even on 64bit windows).

Good luck :-).